.github/workflows/azure-e2e-test.yml

```diff
@@ -0,0 +1,142 @@
+name: azure-e2e-test
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  PODVM_GALLERY_NAME: "podvm_gallery1"
+  PODVM_IMAGE_DEF_NAME: "podvm_image1"
+  PODVM_IMAGE_VERSION: "0.0.1"
+  PODVM_IMAGE_NAME: "peerpod-image-0.0.1"
+  SSH_USERNAME: "peerpod"
+  VM_SIZE: "Standard_D2as_v5"
+
+on:
+  push:
+    branches:
+      - kartikjoshi21/ci-upstream
+
+jobs:
+  build-podvm-image:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: ./.github/actions/setup-azure-build-env
+
+    - name: Clone kata repository
+      run: |
+        git clone https://github.com/kata-containers/kata-containers -b CCv0 ../kata-containers
+
+    - name: Build binaries
+      env:
+        GOPATH: /home/runner/go
+      run: |
+        cd azure/image
+        make binaries
+
+    - uses: azure/login@v1
+      name: 'Az CLI login'
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+
+    - name: Create podvm image
+      id: create-podvm-image
+      env:
+        PKR_VAR_subscription_id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        PKR_VAR_resource_group: ${{ secrets.AZURE_RESOURCE_GROUP }}
+        PKR_VAR_location: ${{ secrets.AZURE_REGION }}
+        PKR_VAR_az_image_name: ${{ env.PODVM_IMAGE_NAME }}
+        PKR_VAR_vm_size: ${{ env.VM_SIZE }}
+        PKR_VAR_ssh_username: ${{ env.SSH_USERNAME }}
+        PKR_VAR_az_gallery_name: ${{ env.PODVM_GALLERY_NAME }}
+        PKR_VAR_az_gallery_image_name: ${{ env.PODVM_IMAGE_DEF_NAME }}
+        PKR_VAR_az_gallery_image_version: ${{ env.PODVM_IMAGE_VERSION }}
+        PKR_VAR_use_azure_cli_auth: "true"
+        GOPATH: /home/runner/go
+        PODVM_DISTRO: "ubuntu"
+      run: |
+        pushd azure/image
+        make image
+        popd
+
+  run-e2e-test:
+    runs-on: ubuntu-latest
+    needs:
+      - build-podvm-image
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: azure/login@v1
+      name: 'Az CLI login'
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+
+    - name: Create provisioner file
+      env:
+        CLUSTER_NAME: "e2e-test-cluster"
+        SSH_KEY_ID: "id_rsa.pub"
+        CLOUD_PROVIDER: "azure"
+        PODVM_IMAGE_NAME: ${{ env.PODVM_IMAGE_NAME }}
+        TEST_E2E_CREATE_RG: "no"
+      run: |
+        cat << EOF > /tmp/provision_azure.properties
+          AZURE_SUBSCRIPTION_ID=${{ secrets.AZURE_SUBSCRIPTION_ID }}
+          AZURE_CLIENT_ID=${{ secrets.AZURE_CLIENT_ID }}
+          AZURE_TENANT_ID=${{ secrets.AZURE_TENANT_ID }}
+          RESOURCE_GROUP_NAME=${{ secrets.AZURE_RESOURCE_GROUP }}
+          CLUSTER_NAME="$CLUSTER_NAME"
+          LOCATION=${{ secrets.AZURE_REGION }}
```

Comment:
note: secrets have to be put buffered in an env before using them in `run` blocks, otherwise you'll run into escaping issues due to special characters. it might work here, because of the EOF literal. but that should be standard practice:

```yaml
env:
 BLA="${{ secrets.bla }}
run: |
  cat << EOF > blub.properties
  BLA="$BLA"
  EOF
```  

also I would not store non-account related values like location and resource_group as secrets, but inline them here as env. that's maybe a matter of preference, but it will make it harder to debug, because those values will be hidden from the output and only users with access to secrets are able to understand what's going on.

Reply:
OH, thats what i was doing earlier. i will change it to use envs.Thanks